src/ucs/sys/iovec.h

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2019.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include <ucs/debug/assert.h>
+
+#include <stdint.h>
+#include <sys/uio.h>
+
+
+/* A direction for copying a data to/from iovec */
+typedef enum ucs_sys_copy_iov_dir {
+    UCS_SYS_COPY_IOV_TO_BUF = 0,
+    UCS_SYS_COPY_BUF_TO_IOV = 1
+} ucs_sys_copy_iov_dir_t;
+
+
+/**
+ * Copy a data from iovec [buffer] to buffer [iovec]
+ *
+ * @param [in] iov           A pointer to an array of iovec elements
+ * @param [in] iov_cnt       A number of elements in a iov array
+ * @param [in] iov_offset    An offset in a iov array
+ * @param [in] buf           A buffer that should be used for copying a data
+ * @param [in] buf_size      A size of a buffer
+ * @param [in] buf_offset    An offset in a buffer
+ * @param [in] dir           Direction that specifies destination and source
+ *
+ * @return The amount, in bytes, of the data that was copied
+ */
+size_t ucs_sys_copy_iov_buf(const struct iovec *iov, size_t iov_cnt,
+                            size_t iov_offset, void *buf, size_t buf_size,
+                            size_t buf_offset, ucs_sys_copy_iov_dir_t dir);
+
+/**
+ * Copy a data from iovec to buffer
+ *
+ * @param [in] iov           A pointer to an array of iovec elements
+ * @param [in] iov_cnt       A number of elements in a iov array
+ * @param [in] iov_offset    An offset in a iov array
+ * @param [in] buf           A buffer that should be used as a destination
+ *                           for copying a data
+ * @param [in] buf_size      A size of a buffer
+ * @param [in] buf_offset    An offset in a buffer
+ *
+ * @return The amount, in bytes, of the data that was copied
+ */
+static inline size_t
+ucs_sys_copy_from_iov(void *buf, size_t buf_size, size_t buf_offset,
```

Comment:
do we really need this wrapper?

Reply:
@yosefe do you mean directly call
`ucs_sys_copy_iov_buf(..., UCS_SYS_IOV_COPY_TO_BUF)` would be better?

Reply:
yes, doesn't seem needed to have extra function just for this param